File:  Validate-Binary-Search-Tree/validate-binary-search-tree.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isValid(node *TreeNode, leftVal int, rightVal int) bool {
	if node == nil {
		return true
	}
	if leftVal < node.Val && node.Val < rightVal {
		return isValid(node.Left, leftVal, node.Val) && isValid(node.Right, node.Val, rightVal)
	} else {
		return false
	}
}
func isValidBST(root *TreeNode) bool {
	if root == nil {
		return true
	}

	// math.INF is not an intin golang so use big number instead
	return isValid(root, -9223372036854775807, 9223372036854775807)
}

File:  Validate-Binary-Search-Tree/validate-binary-search-tree.js
// Adewumi Sunkanmi 2022
// 98. Validate Binary Search Tree
// Medium
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
 var isValidBST = function(root) {
    function valid(node, left, right){
        if(!node){
            return true
        }
        
        if(!(node.val < right && node.val> left)){
            return false
        }
        return (valid(node.left, left, node.val) &&
        valid(node.right,node.val, right))
    }
    
    return valid(root,Number.NEGATIVE_INFINITY, Number.POSITIVE_INFINITY)
    
};